src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -159,11 +159,17 @@ const DotPlotPage = (props) => {
   );
 
   useEffect(() => {
-    if (cellSetsLoading) return;
+    if (cellSetsLoading || !config) return;
+
+    // If using custom genes, check that there are genes in the list
+    if (!config.useMarkerGenes && config.selectedGenes.length === 0) return;
+
+    // If using marker genes, check that the selected number is more than 0
+    if (config.useMarkerGenes && config.nMarkerGenes === 0) return;
 
     // Marker genes calculation needs that the cellIds in groupBy (refer to fn definition)
     // be represented by more than one groups in filterBy to enable comparison
-    if (config?.useMarkerGenes && !hasMoreThanTwoGroupsToCompare) {
+    if (config.useMarkerGenes && !hasMoreThanTwoGroupsToCompare) {
```

Comment:
What is this change for? A bugfix on something else?

Reply:
I made these changes when I was working on the tests and discovered actually`fetchPlotDataWork` was called multiple times when it's not supposed to. These checks prevent that.

Reply:
could you add a structure to those checks by making them into a function named "shouldPlotDataBeFetched" or something like that, have those checks in this function, make the function return a boolean and call that function here?

Reply:
addressed in bd2d5c9